tcmur_cmd_handler.c

```diff
@@ -412,33 +415,19 @@ static int handle_unmap_internal(struct tcmu_device *dev, struct tcmulib_cmd *or
 	}
 state_unlock:
 	/*
-	 * If all calls are successful and nlbas > 0 for all bddls, the
-	 * status should be set to TCMU_STS_ASYNC_HANDLED, or will be the error
-	 * code. If all nlbas = 0 for all bddls, then we can just return
-	 * GOOD status.
-	 */
-	state->status = ret;
-
-	if (ret != TCMU_STS_ASYNC_HANDLED)
-		state->error = true;
-
-	refcount = state->refcount;
+	* unmap_state_put will do the right thing, so always return
+	* TCMU_STS_ASYNC_HANDLED
+	*/
+	if (ret == TCMU_STS_ASYNC_HANDLED) {
+		state->status = TCMU_STS_OK;
+		ret = TCMU_STS_OK;
+	} else {
+		state->status = ret;
+	}
```

Comment:
You need to put the same if/else block in handle_unmap_in_writesame so it does not pass in TCMU_STS_ASYNC_HANDLED to unmap_state_put